### PR TITLE
Params validation for API methods

### PR DESCRIPTION
**Pull request by oakbani**

### CODE REVIEW

File: optimizely/helpers/validator.py [hunk 1]
```diff
@@ -151,3 +152,25 @@ def is_user_profile_valid(user_profile):
       return False
 
   return True
+
+
+def is_non_empty_string(test_var):
```
Comment (aliabbasrizvi) [inline optimizely/helpers/validator.py]: nit. `test_var` is not a good variable name.

File: optimizely/helpers/validator.py [hunk 2]
```diff
@@ -151,3 +152,25 @@ def is_user_profile_valid(user_profile):
...
+
+  Args:
+    test_var: Variable which needs to be validated.
+
+  Returns:
+    Boolean depending upon whether input is valid or not.
+  """
+  PY3 = sys.version_info[0] == 3
+
+  if PY3:
```
Comment (aliabbasrizvi) [inline optimizely/helpers/validator.py]: What happens if you don't do this?

Comment (oakbani) [reply optimizely/helpers/validator.py]: basestring was removed in Python3. https://docs.python.org/3.0/whatsnew/3.0.html

Comment (aliabbasrizvi) [reply optimizely/helpers/validator.py]: I am wondering if there is a better way to accomplish what you are doing here.

File: optimizely/helpers/validator.py [hunk 3]
```diff
@@ -151,3 +152,18 @@ def is_user_profile_valid(user_profile):
...
+def is_non_empty_string(input):
+  """ Determine if provided input is a non-empty string or not.
+
+  Args:
+    input: Variable which needs to be validated.
+
+  Returns:
+    Boolean depending upon whether input is valid or not.
+  """
+  if isinstance(input, string_types) and input:
```
Comment (aliabbasrizvi) [inline optimizely/helpers/validator.py]: input is a built in Python function name. Please change this name.

File: optimizely/helpers/validator.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
 import json
 import jsonschema
 
+from six import string_types
```
Comment (aliabbasrizvi) [inline optimizely/helpers/validator.py]: nit. This is an external package so it should be imported on line 16 and line 17 should be empty.

File: optimizely/helpers/validator.py [hunk 5]
```diff
@@ -151,3 +152,18 @@ def is_user_profile_valid(user_profile):
...
+def is_non_empty_string(input_id_key):
+  """ Determine if provided input_id_key is a non-empty string or not.
+
+  Args:
+    input_id_key: Variable which needs to be validated.
+
+  Returns:
+    Boolean depending upon whether input is valid or not.
+  """
+  if isinstance(input_id_key, string_types) and input_id_key:
```
Comment (aliabbasrizvi) [inline optimizely/helpers/validator.py]: Perhaps flip it to be:

`if input_id_key and isinstance(input_id_key, string_types)`

File: tests/helpers_tests/test_validator.py [hunk 6]
```diff
@@ -130,6 +130,22 @@ def test_is_user_profile_valid__returns_false(self):
...
+
+    self.assertFalse(validator.is_non_empty_string(None))
+    self.assertFalse(validator.is_non_empty_string([]))
+    self.assertFalse(validator.is_non_empty_string({}))
+    self.assertFalse(validator.is_non_empty_string(0))
+    self.assertFalse(validator.is_non_empty_string(99))
+    self.assertFalse(validator.is_non_empty_string(1.2))
+    self.assertFalse(validator.is_non_empty_string(True))
+    self.assertFalse(validator.is_non_empty_string(False))
+    self.assertFalse(validator.is_non_empty_string(""))
```
Comment (aliabbasrizvi) [inline tests/helpers_tests/test_validator.py]: nit. Single quotes

File: tests/helpers_tests/test_validator.py [hunk 7]
```diff
@@ -130,6 +130,22 @@ def test_is_user_profile_valid__returns_false(self):
...
+
+    self.assertTrue(validator.is_non_empty_string("0"))
```
Comment (aliabbasrizvi) [inline tests/helpers_tests/test_validator.py]: nit. Single quotes.

File: tests/helpers_tests/test_validator.py [hunk 8]
```diff
@@ -130,6 +130,22 @@ def test_is_user_profile_valid__returns_false(self):
...
+    self.assertTrue(validator.is_non_empty_string("test_user"))
```
Comment (aliabbasrizvi) [inline tests/helpers_tests/test_validator.py]: Same as above.

**Comment by msohailhussain** [comment]
travis ci is getting failed on Python 3.5, let us investigate this issue.

**Pull request merged**